packages/components/src/calendar/test/index.tsx

```diff
@@ -0,0 +1,751 @@
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).not.toBeDisabled();
+			}
+		} );
+
+		it( 'should support disabling specific days of the week via the `disabled` prop', async () => {
+			const weekendsInMay = [ 3, 4, 10, 11, 17, 18, 24, 25, 31 ];
+			render( <DateCalendar disabled={ { dayOfWeek: [ 0, 6 ] } } /> );
+
+			for ( const date of weekendsInMay ) {
+				expect(
+					getDateButton( new Date( today.getFullYear(), today.getMonth(), date ) )
+				).toBeDisabled();
+			}
+		} );
+
+		it( 'should disable the previous and next months buttons if the `disableNavigation` is set to `true`', async () => {
+			const user = setupUserEvent();
+
+			render( <DateCalendar disableNavigation /> );
+
+			expect( screen.getByRole( 'button', { name: /previous month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+			expect( screen.getByRole( 'button', { name: /next month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+
+			await user.tab();
+			expect( screen.getByRole( 'button', { name: /today/i } ) ).toHaveFocus();
+		} );
+	} );
+
+	// Note: we're not testing localization of strings. We're only testing
+	// that the date formatting, computed dir, and calendar format are correct.
+	describe( 'Localization', () => {
+		it( 'should localize the calendar based on the `locale` prop', async () => {
+			const user = setupUserEvent();
+
+			render( <DateCalendar locale={ ar } /> );
+
+			// Check computed writing direction
+			expect( screen.getByRole( 'application', { name: 'Date calendar' } ) ).toHaveAttribute(
+				'dir',
+				'rtl'
+			);
+
+			// Check month name
+			const grid = screen.getByRole( 'grid', {
+				name: monthNameFormatter( 'ar' ).format( today ),
+			} );
+			expect( grid ).toBeInTheDocument();
+
+			// Check today button
+			expect( getDateButton( today, {}, 'ar' ) ).toHaveAccessibleName( /today/i );
+
+			await user.tab();
+			await user.tab();
+			await user.tab();
+			expect( getDateButton( today, {}, 'ar' ) ).toHaveFocus();
+
+			await user.keyboard( '{Home}' );
+			expect( getDateButton( startOfWeek( today, { locale: ar } ), {}, 'ar' ) ).toHaveFocus();
+		} );
+
+		it( 'should support timezones according to the `timeZone` prop', async () => {
+			const user = setupUserEvent();
+			const onSelect = jest.fn();
+
+			render( <DateCalendar timeZone="Asia/Tokyo" onSelect={ onSelect } /> );
+
+			// For someone in Tokyo, 20:00 UTC is the next day.
+			expect( getDateButton( tomorrow ) ).toHaveAccessibleName( /today/i );
+
+			// Select tomorrow's button (which is today in Tokyo)
+			const tomorrowButton = getDateButton( tomorrow );
+			await user.click( tomorrowButton );
+			expect( onSelect ).toHaveBeenCalledWith(
+				fromZonedTime( tomorrow, 'Asia/Tokyo' ),
+				fromZonedTime( tomorrow, 'Asia/Tokyo' ),
+				expect.objectContaining( { today: true } ),
+				expect.objectContaining( { type: 'click', target: tomorrowButton } )
+			);
```

Comment:
Can't we do something like this using the built-in `addHours` to prevent from using `date-fns-tz`?

```suggestion
			const tokyoOffset = new TZDate( tomorrow, 'Asia/Tokyo' ).getTimezoneOffset() / 60;
			const tokyoTomorrow = addHours( tomorrow, 1 * tokyoOffset );
			expect( onSelect ).toHaveBeenCalledWith(
				new TZDate( tokyoTomorrow, 'Asia/Tokyo' ),
				new TZDate( tokyoTomorrow, 'Asia/Tokyo' ),
				expect.objectContaining( { today: true } ),
				expect.objectContaining( { type: 'click', target: tomorrowButton } )
			);
```

Reply:
Good call, I'll refactor away from `date-fns-tz` and use your proposed solution